pkg/roachprod/vm/vm.go

```diff
@@ -140,6 +162,8 @@ const (
 type CreateOpts struct {
 	ClusterName    string
 	Lifetime       time.Duration
+	CustomLabelMap map[string]string
```

Comment:
Nit: in the `VM` type the field is simply `Labels`. Here you're using a `Map` suffix which is inconsistent naming. I think this should be named `CustomLabels` for consistency.